MatrixSDK/MXSession.m

```diff
@@ -4072,4 +4106,33 @@ - (void)markOutdatedPublicisedGroupsByUserData
     return publicisedGroups;
 }
 
+#pragma mark - Virtual Rooms
+
+- (void)setVirtualRoom:(NSString *)virtualRoomId forNativeRoom:(NSString *)nativeRoomId
+{
+    [self setVirtualRoom:virtualRoomId forNativeRoom:nativeRoomId notify:YES];
+}
+
+- (void)setVirtualRoom:(NSString *)virtualRoomId forNativeRoom:(NSString *)nativeRoomId notify:(BOOL)notify
+{
+    if (virtualRoomId)
+    {
+        nativeToVirtualRoomIds[nativeRoomId] = virtualRoomId;
+    }
+    else
+    {
+        [nativeToVirtualRoomIds removeObjectForKey:nativeRoomId];
+    }
+    
+    //  post an update of the virtual rooms.
+    [[NSNotificationCenter defaultCenter] postNotificationName:kMXSessionVirtualRoomsDidChangeNotification
```

Comment:
It should be under the notify `condition`

Reply:
Ah yeah, updated